### ISSUE TITLE
BUG: False positive warning on a query key implementation that uses query factories method, or when passing queryKey as a shorthand property

### ISSUE THREAD

**Issue by onyedikachi23** [tigerabrodi/eslint-plugin-react-query-keys#1]
I build query factories for parts of my code using the [queryOptions](https://tanstack.com/query/latest/docs/framework/react/reference/queryOptions#queryoptions) API according to this guideline and pattern from [TkDodo's blog](https://tkdodo.eu/blog/the-query-options-api#query-factories).

Here's an example implementation of mine: 

```ts
export const jobsQueries = {
	// Base key for all job-related queries
	all: () => ["jobs"] as const,

	// Plural base key for searches
	searches: () => [...jobsQueries.all(), "search"] as const,

	// Singular query for searching jobs with pagination
	search: ({
		page = DEFAULT_PAGE_NUMBER,
		limit = DEFAULT_PAGE_SIZE,
		search,
	}: SearchQueryPaginatedParams) =>
		queryOptions({
			queryKey: [
				...jobsQueries.searches(),
				{ page, limit, search },
			] as const,
			queryFn: () => fetchSearchJobs({ page, limit, search }),
		}),

	// Plural base key for tracking jobs
	trackings: () => [...jobsQueries.all(), "tracking"] as const,

	// Singular query for tracking jobs with pagination
	tracking: ({
		page = DEFAULT_PAGE_NUMBER,
		limit = DEFAULT_PAGE_SIZE,
	}: PaginationParams) =>
		queryOptions({
			queryKey: [...jobsQueries.trackings(), { page, limit }] as const,
			queryFn: () => fetchTrackingJobs({ page, limit }),
		}),

	// Plural base key for job details
	details: () => [...jobsQueries.all(), "detail"],

	// Singular query for a specific job by ID
	detail: (jobId: string) =>
		queryOptions({
			queryKey: [...jobsQueries.details(), jobId] as const,
			queryFn: () => fetchJobDetails(jobId),
		}),
};

```

However, when using `jobQueries.all()` for a queryKey, I get this warning:

```txt
Avoid using raw arrays or strings for query keys. Use query key factories instead.
```

Another case is passing queryKey as a shorthand property, I still get the same warning:

```ts

await queryClient.cancelQueries({ queryKey })

```

**Comment by tigerabrodi** [comment]
@onyedikachi23 

this is great

ill make sure we handle the queries pattern too

i think this could be a different rule

so if people wanna decide to handle the queries approach or query keys approach

ill implement it

**Comment by tigerabrodi** [comment]
in hindsight, ill just do it as the same rule ✅

**Comment by onyedikachi23** [comment]
Thanks man. I've only used it for a few days now, but it has helped me spot inconsistencies a lot already.

I think GitHub should leave a tag here when you push the update.

**Comment by tigerabrodi** [comment]
@onyedikachi23 should be fixed

please upgrade to `eslint-plugin-react-query-keys@0.1.0`
